File: Python/TurnTicketDispenser/turn_ticket.py
class TurnTicket(object):
    def __init__(self, turnNumber):
        self.turnNumber = turnNumber

class TurnNumberSequence(object):
    _turnNumber = -1

    @staticmethod
    def next_turn_number():
        TurnNumberSequence._turnNumber += 1
        return TurnNumberSequence._turnNumber


class TicketDispenser(object):
    def getTurnTicket(self):
        newTurnNumber = TurnNumberSequence.next_turn_number()
        newTurnTicket = TurnTicket(newTurnNumber)
        return newTurnTicket

File: Ruby/TurnTicketDispenser/turn_ticket.rb
class TurnTicket
  def initialize turn_number
    @turn_number = turn_number
  end
end

class TurnNumberSequence

  @@turn_number = -1
  
  def self.next_turn_number
    @@turn_number += 1

    @@turn_number
  end
end

class TicketDispenser
  def get_turn_ticket
    new_turn_number = TurnNumberSequence.next_turn_number
    new_turn_ticket = TurnTicket.new(new_turn_number)

    new_turn_ticket
  end
end
